### PR TITLE
EE 6374 Replace FrequencyCalculator Spock tests with JUnit

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Replaced all the Groovy/Spock tests for FrequencyCalculator with equivalent JUnit ones. Also added two new tests to ensure FrequencyCalculator returns the expected response when there are duplicate payments for a given month or week.

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        IncomeRecord incomeRecord = incomeRecordForDateWithMonthPayNumber(dates);
+
+        Frequency frequency = calculate(incomeRecord);
+
+        assertThat(frequency).isEqualTo(CALENDAR_MONTHLY);
+    }
+
+    @Test
+    public void shouldReturnMonthlyWhen2SameDateConsectutiveMonthsWithMonthNumberPresent() {
+        List<LocalDate> dates = generateWeeklyDatesFrom(LocalDate.of(2017, Month.DECEMBER, 1))
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java]: The old test appears to generate monthly payments?
```groovy
def dates = Stream.iterate(LocalDate.of(2017, Month.DECEMBER, 1), { date -> date.plusMonths(-1)}
```

File: src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+        IncomeRecord incomeRecord = incomeRecordForDateWithMonthPayNumber(dates);
+
+        Frequency frequency = calculate(incomeRecord);
+
+        assertThat(frequency).isEqualTo(CALENDAR_MONTHLY);
+    }
+
+    @Test
+    public void shouldReturnMonthlyWhen1MonthWithMonthNumberPresent() {
+        List<LocalDate> dates = generateWeeklyDatesFrom(LocalDate.of(2017, Month.DECEMBER, 1))
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java]: Old test had monthly dates?

File: src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+    private Stream<LocalDate> generateCalendarMonthlyDatesFrom(LocalDate startDate) {
+        return Stream.iterate(startDate, date -> date.minusMonths(1));
+    }
+
+    private IncomeRecord incomeRecordForDate(List<LocalDate> dates) {
+        List<Income> paye = dates.stream()
+            .map(date -> new Income(BigDecimal.ONE, date, null, null, "some employer ref"))
+            .collect(Collectors.toList());
+
+        return new IncomeRecord(paye, Lists.emptyList(), Lists.emptyList(), null);
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java]: I noticed that earlier Collections.emptyList() was used...  this should be consistent.  Tim likes to import statics, which slims down the code and makes it harder to make such inconsistencies... might be worth doing.

**Review by mikehalmaho — CHANGES_REQUESTED** [review]

**Review by mikehalmaho — APPROVED** [review]
